Modules/posixmodule.c

```diff
@@ -10015,6 +10020,174 @@ os_times_impl(PyObject *module)
 #endif /* HAVE_TIMES */
 
 
+#if defined(HAVE_TIMERFD_CREATE)
+#define ONE_SECOND_IN_NS (1000 * 1000 * 1000)
+#define EXTRACT_NSEC(value)  (long)( ( (double)(value) - (time_t)(value) ) * 1e9)
+
+static PyObject *
+build_itimerspec(const struct itimerspec* curr_value)
+{
+    return PyTuple_Pack(
+        2,
+        PyFloat_FromDouble((double)(curr_value->it_interval.tv_sec) + (double)curr_value->it_interval.tv_nsec * 1e-9),
+        PyFloat_FromDouble((double)(curr_value->it_value.tv_sec) + (double)curr_value->it_value.tv_nsec * 1e-9)
+    );
+}
+
+static PyObject *
+build_itimerspec_ns(const struct itimerspec* curr_value)
+{
+    return PyTuple_Pack(
+        2,
+        PyLong_FromLongLong((long long)curr_value->it_interval.tv_sec * ONE_SECOND_IN_NS + (long long)curr_value->it_interval.tv_nsec),
+        PyLong_FromLongLong((long long)curr_value->it_value.tv_sec * ONE_SECOND_IN_NS + (long long)curr_value->it_value.tv_nsec)
+    );
+}
+
+/*[clinic input]
+os.timerfd_create
+
+    clockid: int
+    flags: int
+
```

Comment:
Make parameters positional-only?
```suggestion
    /

```

Reply:
@serhiy-storchaka 
Updated.